src/main/java/com/devonfw/quarkus/productmanagement/logic/UcFindProduct.java

```diff
@@ -1,24 +1,61 @@
 package com.devonfw.quarkus.productmanagement.logic;
 
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.transaction.Transactional;
+
 import org.springframework.data.domain.Page;
 
+import com.devonfw.quarkus.productmanagement.domain.model.ProductEntity;
+import com.devonfw.quarkus.productmanagement.domain.repo.ProductRepository;
+import com.devonfw.quarkus.productmanagement.service.v1.mapper.ProductMapper;
 import com.devonfw.quarkus.productmanagement.service.v1.model.ProductDto;
 import com.devonfw.quarkus.productmanagement.service.v1.model.ProductSearchCriteriaDto;
 
-public interface UcFindProduct {
-  Page<ProductDto> findProducts(ProductSearchCriteriaDto dto);
-
-  Page<ProductDto> findProductsByCriteriaApi(ProductSearchCriteriaDto dto);
-
-  Page<ProductDto> findProductsByQueryDsl(ProductSearchCriteriaDto dto);
-
-  Page<ProductDto> findProductsByTitleQuery(ProductSearchCriteriaDto dto);
-
-  Page<ProductDto> findProductsByTitleNativeQuery(ProductSearchCriteriaDto dto);
-
-  Page<ProductDto> findProductsOrderedByTitle();
-
-  ProductDto findProduct(String id);
-
-  ProductDto findProductByTitle(String title);
-}
+@Named
+@Transactional
+public class UcFindProduct {
+  @Inject
+  ProductRepository ProductRepository;
+
+  @Inject
+  ProductMapper mapper;
+
+  public Page<ProductDto> findProducts(ProductSearchCriteriaDto searchCriteria) {
+
+    Page<ProductEntity> products = this.ProductRepository.findProducts(searchCriteria);
+    if (products.isEmpty()) {
+      return null;
```

Comment:
in methods that return collections or pages, we dont want to return null, instead an empty page should be returned

Reply:
fine. Thanks.